compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -2993,7 +2993,16 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         {
             (s, " +")
         } else {
-            (span.shrink_to_hi(), ":")
+            (
+                if let GenericParamKind::Type { default: Some(default), synthetic: _ } = param.kind
+                {
+                    span.data().with_hi(default.span.lo() - rustc_span::BytePos(3))
```

Comment:
What is this `- 3`?

Reply:
```
2 |     struct StructA<A, B = A> {
  |                       ^^^^^ required by the implicit `Sized` requirement on this type parameter in `StructA`
```

Because there didn't seem to be any value to specify "=".

GenericParamKind::Type::default.span.lo is begin A of BytePos
and
param.span.hi eq GenericParamKind::Type::default.span.lo .

Reply:
So this doesn't work for cases when there are a lot of spaces between `=`?

Reply:
That's right. I'll fix it and repost. thank you.